boto/auth.py

```diff
@@ -1009,27 +1013,6 @@ def get_auth_handler(host, config, provider, requested_capability=None):
     return ready_handlers[-1]
 
 
-def detect_potential_sigv4(func):
```

Comment:
I don't think anything should be using this, but let's not remove it.

You can mark this as deprecated or raise a deprecation warning, but I'm hesitant to remove it outright without any warning.